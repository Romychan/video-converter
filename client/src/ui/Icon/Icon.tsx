import clsx from 'clsx';

import IconsSVG from '../../assets/img/general/icons.svg';

export interface IIcon {
  name: string;
  color?: string;
  size?: number;
  className?: string;
}

export const Icon = ({
  name,
  color = '',
  size = 16,
  className = '',
}: IIcon) => (
  <svg
    className={clsx(`icon icon-${name}`, className)}
    stroke={color}
    width={size}
    height={size}
    data-testid="icon"
  >
    <use xlinkHref={`${IconsSVG}#${name}`} />
  </svg>
);
