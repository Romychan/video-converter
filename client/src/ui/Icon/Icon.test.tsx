import { RenderResult, render, screen } from '@testing-library/react';

import { Icon } from './Icon';

describe('Icon', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = render(<Icon name="test" size={18} color="#ffffff" />);
  });

  it('should render correctly', () => {
    const icon = screen.getByTestId('icon');

    expect(icon).toBeInTheDocument();
    expect(icon.getAttribute('class')).toMatch(/icon-test/gi);
    expect(icon.getAttribute('stroke')).toMatch(/#ffffff/gi);
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
