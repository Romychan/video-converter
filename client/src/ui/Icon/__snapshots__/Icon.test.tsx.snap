// Vitest Snapshot v1

exports[`Icon > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        class="icon icon-test"
        data-testid="icon"
        height="18"
        stroke="#ffffff"
        width="18"
      >
        <use
          xlink:href="/src/assets/img/general/icons.svg#test"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="icon icon-test"
      data-testid="icon"
      height="18"
      stroke="#ffffff"
      width="18"
    >
      <use
        xlink:href="/src/assets/img/general/icons.svg#test"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
